src/Span.h

```diff
@@ -1375,4 +1431,206 @@ void Span::serialize_json(RapidJsonWriter &writer) const
     writer.EndObject();
 }
 
-} // namespace zipkin
\ No newline at end of file
+namespace __impl
+{
+
+struct Span2 {
+    enum Kind {
+        UNKNOWN,
+        CLIENT,
```

Comment:
PRODUCER and CONSUMER are also kinds, btw. sadly the conversion code got a bit longer to support these https://github.com/openzipkin/zipkin/blob/master/zipkin/src/main/java/zipkin/internal/Span2Converter.java#L300